modules/custom/az_paragraphs/src/Plugin/Field/FieldFormatter/AZBackgroundMediaFormatter.php

```diff
@@ -0,0 +1,627 @@
+                property. For more information on the css background property see
+                http://www.w3schools.com/css/css_background.asp"'
+      ),
+    ];
+
+    // The z-index property.
+    $form['css_settings']['z_index'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Z Index'),
+      '#description' => $this->t(
+        'The z-index property specifies the stack order of an element. An element with greater stack order is
+                      always in front of an element with a lower stack order. Note: z-index only works on positioned
+                      elements (position:absolute, position:relative, or position:fixed)'
+      ),
+      '#default_value' => $settings['css_settings']['z_index'],
+    ];
+
+    // The background-color property.
+    $form['css_settings']['color'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Color'),
+      '#description' => $this->t(
+        'The background color formatted as any valid css color format (e.g. hex, rgb, text, hsl)
+                      [<a href="@url">css property: background-color</a>].',
+        ['@url' => 'https://developer.mozilla.org/en-US/docs/Web/CSS/linear-gradient']
+      ),
+      '#default_value' => $settings['css_settings']['color'],
+    ];
+
+    // The background-size x property.
+    $form['css_settings']['x'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Horizontal Alignment'),
+      '#description' => $this->t(
+        'The horizontal alignment of the background image formatted as any valid css alignment.
+                      [<a href="http://www.w3schools.com/css/pr_background-position.asp">
+                      css property: background-position
+                      </a>]'
+      ),
+      '#default_value' => $settings['css_settings']['x'],
+    ];
+    // The background-size y property.
+    $form['css_settings']['y'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Vertical Alignment'),
+      '#description' => $this->t(
+        'The vertical alignment of the background image formatted as any valid css alignment.
+                      [<a href="http://www.w3schools.com/css/pr_background-position.asp">
+                      css property: background-position
+                      </a>]'
+      ),
+      '#default_value' => $settings['css_settings']['y'],
+    ];
+    // The background-attachemetn property.
+    $form['css_settings']['attachment'] = [
+      '#type' => 'radios',
+      '#title' => $this->t('Background Attachment'),
+      '#description' => $this->t(
+        'The attachment setting for the background image.
+                      [<a href="http://www.w3schools.com/css/pr_background-attachment.asp">
+                      css property: background-attachment
+                      </a>]'
+      ),
+      '#options' => [
+        FALSE => $this->t('Ignore'),
+        'scroll' => 'Scroll',
+        'fixed' => 'Fixed',
+      ],
```

Comment:
Nitpick: I think it would be preferable to use `#empty_option` rather than having a `FALSE` option (the empty value should still evaluate as `FALSE` for the Twig `if` statements).  We should also be passing the other option strings through `$this->t()`:
```suggestion
      '#options' => [
        'scroll' => $this->t('Scroll'),
        'fixed' => $this->t('Fixed'),
      ],
      '#empty_option' => $this->t('Ignore'),
```